### ISSUE TITLE
Syncs don't trigger immediately after a partial reset

### ISSUE THREAD

**Issue by nataliekwong** [airbytehq/airbyte#32636]
If I trigger a reset that is partial (some streams or a single stream), I have to wait for the Reset to complete, and then click "Sync now" in order for a sync to occur after the reset.

If I trigger a full reset from the replication settings, which resets all my streams, a sync will be started immediately afterwards. This is what I would expect because I still want my data, not empty tables. 

This is a confusing experience because I expect the same thing to happen when I initiate a reset - that the reset runs, then triggers a sync immediately after.

**Comment by nataliekwong** [comment]
I believe this is related behavior: when a stream is reset, the subsequent sync should only sync the stream that was reset, not all the enabled streams for that connection: 

> resetting a single stream in the UI runs T&D for all streams in the sync catalog. This shouldn't happen

Relevant Slack thread: https://airbytehq-team.slack.com/archives/C02U6CXJQFN/p1700337205175069

**Comment by davinchia** [comment]
Summary here: https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1701202345616629

TLDR: All resets do this today. We won't change this behaviour is Q4B. We might revisit in Q1.
